### PR TITLE
feat(ui5-busy-indicator): add new "delay" property

### PR DESCRIPTION
**Pull request by TeodorTaushanov**
Also added new "busyDelay" property to ui5-list, ui5-table, ui5-li-notification-group, ui5-li-notification.

### CODE REVIEW

File: packages/fiori/src/NotificationListGroupItem.hbs [hunk 1]
```diff
@@ -82,6 +82,9 @@
 	</ui5-list>
 
 	{{#if busy}}
-		<ui5-busy-indicator active size="Medium" class="ui5-nli-busy"></ui5-busy-indicator>
+		<ui5-busy-indicator
+				busyDelay="{{delay}}"
+				active size="Medium"
+				class="ui5-nli-busy"></ui5-busy-indicator>
```
Comment (fifoosid) [inline packages/fiori/src/NotificationListGroupItem.hbs:88]: Please put the every attribute on new line & the closing tag on new line as well

File: packages/fiori/src/NotificationListItem.hbs [hunk 2]
```diff
@@ -96,6 +96,10 @@
 	</div>
 
 	{{#if busy}}
-		<ui5-busy-indicator active size="Medium" class="ui5-nli-busy"></ui5-busy-indicator>
+		<ui5-busy-indicator
+				busyDelay="{{delay}}"
+				active
+				size="Medium"
+				class="ui5-nli-busy"></ui5-busy-indicator>
```
Comment (fifoosid) [inline packages/fiori/src/NotificationListItem.hbs:103]: Same here

File: packages/main/src/List.js [hunk 3]
```diff
@@ -185,6 +186,18 @@ const metadata = {
 			type: Boolean,
 		},
 
+		/**
+		 * Defines the delay in milliseconds, after which the busy indicator will show up for this control.
```
Comment (fifoosid) [inline packages/main/src/List.js:190]: for this component

File: packages/main/test/specs/BusyIndicator.spec.js [hunk 4]
```diff
@@ -18,19 +18,6 @@ describe("BusyIndicator general interaction", () => {
 		assert.strictEqual(input.getProperty("value"), "0", "itemClick is not thrown");
 	});
 
-	it("tests focus handling", () => {
```
Comment (fifoosid) [inline packages/main/test/specs/BusyIndicator.spec.js:21]: It might be a good idea to keep this test

**Review by fifoosid — DISMISSED** [review]
You can add tests for the new use cases. Some minor comments to be addressed as well.

**Review by fifoosid — APPROVED** [review]
Waiting for review from @SAP/ui5-webcomponents-topic-rd and we can merge it

File: packages/main/src/BusyIndicator.js [hunk 5]
```diff
@@ -78,6 +79,27 @@ const metadata = {
...
+		 * @defaultValue 1000
+		 * @public
+		 */
+		delay: {
+			type: Integer,
+			defaultValue: 1000,
+		},
+
+		/**
+		 * Defines if the control is currently in busy state.
```
Comment (dimovpetar) [inline packages/main/src/BusyIndicator.js:96]: "control" is not appropriate term here. Replace it with "component"

File: packages/main/src/BusyIndicator.js [hunk 6]
```diff
@@ -188,6 +210,23 @@ class BusyIndicator extends UI5Element {
...
+		if (this.active) {
+			if (!this._isBusy && !this._busyTimeoutId) {
+				this._busyTimeoutId = setTimeout(() => {
+					delete this._busyTimeoutId;
+					this._isBusy = true;
+				}, Math.max(0, this.delay));
+			}
+		} else {
+			if (this._busyTimeoutId) {
+				clearTimeout(this._busyTimeoutId);
```
Comment (dimovpetar) [inline packages/main/src/BusyIndicator.js:223]: also clearTimeout onExitDOM

**Review by dimovpetar — CHANGES_REQUESTED** [review]

File: packages/fiori/src/NotificationListGroupItem.hbs [hunk 7]
```diff
@@ -82,6 +82,11 @@
 	</ui5-list>
 
 	{{#if busy}}
-		<ui5-busy-indicator active size="Medium" class="ui5-nli-busy"></ui5-busy-indicator>
+		<ui5-busy-indicator
+				busyDelay="{{delay}}"
```
[the remark this answers is not in the document]
Comment (TeodorTaushanov) [reply packages/fiori/src/NotificationListGroupItem.hbs]: Done.

File: packages/main/src/BusyIndicator.js (hunk 5 above)
Comment (TeodorTaushanov) [reply packages/main/src/BusyIndicator.js]: Done.

**Review by dimovpetar — APPROVED** [review]

**Review by fifoosid — APPROVED** [review]

**Pull request merged**